File: problems_java/leetCode24.java
package problems_java;
 public class ListNode {
         int val;
         ListNode next;
         ListNode() {}
         ListNode(int val) { this.val = val; }
         ListNode(int val, ListNode next) { this.val = val; this.next = next; }
    }
public class leetCode24 {
    public stattic ListNode(ListNode head){
        ListNode dummy = new ListNode(0);
        dummy.next = head;
        ListNode prev = dummy;
        while(prev.next != null && prev.next.next != null){
             ListNode cur = prev.next;
             ListNode post = prev.next.next;

             // start the swap
             cur.next = post.next;
             post.next = cur;
             prev = post;

             prev = prev.next.next;
        }
        return dummy.next;
    }
    public static void main(String[] args){
    }
}

File: mid/leetCode24.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def swapPairs(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        dummyNode = ListNode(0)
             
        pre = dummyNode
         
        while pre.next and pre.next.next:
            cur = pre.next 
            post = pre.next.next 

            # pre cur post   swap
            cur.next = post.next 
            post.next = cur 
            pre.next = post 

            pre = pre.next.next 

        return dummyNode.next 
